core/variant/variant_internal.h

```diff
@@ -46,6 +46,9 @@ class VariantInternal {
 			case Variant::AABB:
 				init_aabb(v);
 				break;
+			case Variant::BASIS:
+				init_basis(v);
+				break;
```

Comment:
Nitpick, but since you need to amend the commit to sync the class reference:

Could you reorder the first `case`s here to match the order of `Variant::Type`, which is also the order of declaration of the `init_` methods? It should be:
```
STRING
TRANSFORM2D
AABB
BASIS
TRANSFORM
STRING_NAME
etc. the rest is correct
```

And while at it I reviewed the list to make sure we don't miss any. Could you add this comment where the `init_` methods are declared?
```diff
diff --git a/core/variant/variant_internal.h b/core/variant/variant_internal.h
index 7d33d85cd6..c8ad1554fe 100644
--- a/core/variant/variant_internal.h
+++ b/core/variant/variant_internal.h
@@ -192,11 +192,14 @@ public:
                v->type = GetTypeInfo<T>::VARIANT_TYPE;
        }
 
+       // Should be in the same order as Variant::Type for consistency.
+       // Those primitive and vector types don't need an `init_` method:
+       // Nil, bool, float, Vector2/i, Rect2/i, Vector3/i, Plane, Quat, Color, RID.
+       // Object is a special case, handled via `object_assign_null`.
        _FORCE_INLINE_ static void init_string(Variant *v) {
                memnew_placement(v->_data._mem, String);
                v->type = Variant::STRING;
        }
-
        _FORCE_INLINE_ static void init_transform2d(Variant *v) {
                v->_data._transform2d = memnew(Transform2D);
                v->type = Variant::TRANSFORM2D;
```

Reply:
CC @vnen to review my added comment as I'm not a Variant expert :)

Reply:
Confirmed that the added comment is correct :)
```
<vnen> yeah, those types don’t allocate extra memory so they don’t need initialization, only setting the type is enough
```